data-prepper-plugins/dynamodb-source-coordination-store/src/main/java/org/opensearch/dataprepper/plugins/sourcecoordinator/dynamodb/DynamoDbSourcePartitionItem.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sourcecoordinator.dynamodb;
+
+import org.opensearch.dataprepper.model.source.coordinator.SourcePartitionStatus;
+import org.opensearch.dataprepper.model.source.coordinator.SourcePartitionStoreItem;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;
+
+import java.time.Instant;
+
+@DynamoDbBean
+public class DynamoDbSourcePartitionItem implements SourcePartitionStoreItem {
+
+    private String sourcePartitionKey;
+    private String partitionOwner;
+    private Long version;
```

Comment:
Question: What is the version here? It's not part of the interface

Reply:
This version is a concept specific to DynamoDb, which is why it is not part of the interface, and we will use it for optimistic locking (https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBMapper.OptimisticLocking.html)